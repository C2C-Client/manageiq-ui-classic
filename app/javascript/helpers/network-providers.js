import handleFailure from './handle-failure';
import { API } from '../http_api';

export const networkProviders = (options = {}) => {
  const newOptions = {
    attributes: ['id', 'name'],
    handleFailure,
    ...options,
  };
  const baseUrl = '/api/providers?collection_class=ManageIQ::Providers::NetworkManager&expand=resources';
  const url = `${baseUrl}&attributes=${newOptions.attributes.map(encodeURIComponent).join(',')}`;

  return API.get(url)
    .then(response => response.resources || [])
    .catch(newOptions.handleFailure);
};
