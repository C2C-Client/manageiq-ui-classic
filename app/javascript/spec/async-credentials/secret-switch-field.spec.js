import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SecretSwitchField from '../../components/async-credentials/secret-switch-field';
import { FieldProviderComponent as FieldProvider } from '../helpers/fieldProvider';

const DummyComponent = ({
  isDisabled,
  validateOnMount, // eslint-disable-line
  validate, // eslint-disable-line
  editMode, // eslint-disable-line
  buttonLabel,
  setEditMode,
  ...props
}) => <button {...props} onClick={setEditMode} disabled={isDisabled} type="button">{buttonLabel || 'Dummy'}</button>;

describe('Secret switch field component', () => {
  let initialProps;
  let changeSpy;
  let getStateSpy;
  beforeEach(() => {
    changeSpy = jest.fn();
    getStateSpy = jest.fn().mockReturnValue({
      values: {},
    });
    initialProps = {
      FieldProvider,
      edit: false,
      formOptions: {
        renderForm: ([secret]) => <DummyComponent {...secret} />,
        change: changeSpy,
        getState: getStateSpy,
      },
    };
  });

  afterEach(() => {
    changeSpy.mockReset();
  });

  it('should render correctly in non edit mode', () => {
    const wrapper = mount(<SecretSwitchField {...initialProps} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly in edit mode', () => {
    const wrapper = mount(<SecretSwitchField {...initialProps} edit />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Hooks are not supported in enzyme jest.
   * Instead of adding another testing utilities we can wait for while until its added or until we know enzyme will not support hooks
   * and we can use another library
   * https://github.com/airbnb/enzyme/issues/2011
   */
  it('should render correctly switch to editing', () => {
    const wrapper = mount(<SecretSwitchField {...initialProps} edit />);
    wrapper.find('button').simulate('click');
    expect(changeSpy).toHaveBeenCalled();
  });
});
