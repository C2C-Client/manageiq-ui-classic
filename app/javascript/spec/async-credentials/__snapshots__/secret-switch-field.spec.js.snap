// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Secret switch field component should render correctly in edit mode 1`] = `
<SecretSwitchField
  FieldProvider={[Function]}
  cancelEditLabel="Cancel"
  changeEditLabel="Change"
  edit={true}
  formOptions={
    Object {
      "change": [MockFunction],
      "getState": [MockFunction],
      "renderForm": [Function],
    }
  }
  isDisabled={false}
>
  <DummyComponent
    buttonLabel="Change"
    component="credentials-password-edit"
    editMode={true}
    isDisabled={false}
    placeholder="●●●●●●●●"
    setEditMode={[Function]}
    type="password"
    validateOnMount={true}
  >
    <button
      component="credentials-password-edit"
      disabled={false}
      onClick={[Function]}
      placeholder="●●●●●●●●"
      type="button"
    >
      Change
    </button>
  </DummyComponent>
</SecretSwitchField>
`;

exports[`Secret switch field component should render correctly in non edit mode 1`] = `
<SecretSwitchField
  FieldProvider={[Function]}
  cancelEditLabel="Cancel"
  changeEditLabel="Change"
  edit={false}
  formOptions={
    Object {
      "change": [MockFunction],
      "getState": [MockFunction],
      "renderForm": [Function],
    }
  }
  isDisabled={false}
>
  <DummyComponent
    component="text-field"
    isDisabled={false}
    type="password"
    validate={false}
    validateOnMount={true}
  >
    <button
      component="text-field"
      disabled={false}
      type="button"
    >
      Dummy
    </button>
  </DummyComponent>
</SecretSwitchField>
`;
